docs/architecture/api-credentials-flows.md

```diff
@@ -0,0 +1,18 @@
+# Overview
+
+This document describes how [Application](/docs/terminology.md#Application) provides and updates credentials for [API Definition](/docs/terminology.md#Application-API-Definiton) per [Runtime](/docs/terminology.md#Runtime).
+
+## Details
+This diagram illustrates the API credentials flow in details: 
+
+>**NOTE:** The Application provides [Webhook API](/docs/terminology.md#Application-Webhook-API) where [Management Plane](/docs/terminology.md#Management-Plane) will ask for new credentials for given API Definition.
+
+>**NOTE:** The Application or [Administrator](/docs/terminology.md#Administrator) can exchange credentials for registered APIs at any time.
+
+![Application Webhook](./assets/api-credentials-flow.svg)
+
+0. Assume we have Application with one API Definiton which is already registered into Management Plane. No Runtimes assigned yet.
+1. The Administrator requests new Runtime.
+2. The Management Plane creates Runtime with configuration for [Agent](/docs/terminology.md#Runtime-Agent) and asks Application for new credentials. 
+3. The Application sets API Definition credentials for given Runtime.
+4. The Agent enables Runtime to call Application APIs.
```

Comment:
do we want to describe how the flow will look like, when Application does not implement Webhooks?

Reply:
Just Added as a note, no diagram for that flow.